### PR TITLE
Added timezone check for Personal Calendar web part

### LINKED ISSUE #166: Personal Calendar Web Part time not local
Category
- [ ] Question
- [  ] Bug
- [ ] Enhancement

#### Expected or Desired Behavior

Personal Calendar web part to display exchange appointments/meetings based on the current user in local time

#### Observed Behavior

The calendar web part is displaying meetings based on the current user but the time is off by -1 hour. Example: Meeting in outlook is set for 14:30, the web part will display this as 13:30. Not sure if this is set to UTC time only.

Based in the UK we have daylight savings time so are currently -1 hour behind UTC.

#### Steps to Reproduce

Add calendar web part to a page 

Thanks for your contribution! Sharing is caring.

image

image

### PR DESCRIPTION
**Pull request by Laskewitz**
#### Category
- [ ] Bug Fix
- [x] New Feature
- Related issues: fixes #166 

#### What's in this Pull Request?

Added timezone check for personal calendar webpart.

### CODE REVIEW

File: solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx [hunk 1]
```diff
@@ -21,6 +22,24 @@ export default class PersonalCalendar extends React.Component<IPersonalCalendarP
...
+  /**
+   * Get timezone for logged in user
+   */
+  private _getTimeZone() {
+    return new Promise<string>((resolve, reject) => {
+      this.props.graphClient
+        // get the mailbox settings
+        .api(`me/mailboxSettings`)
+        .version("v1.0")
+        .get((err: any, res: microsoftgraph.MailboxSettings): void => {
```
Comment (waldekmastykarz) [inline solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx]: Here we're missing handling error. If the call fails, the promise is not rejected and the web part keeps indicating that the data is loading.

File: solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx [hunk 2]
```diff
@@ -21,6 +22,24 @@ export default class PersonalCalendar extends React.Component<IPersonalCalendarP
     };
   }
 
+
```
Comment (waldekmastykarz) [inline solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx]: No need for the extra empty lines

File: solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx [hunk 3]
```diff
@@ -21,6 +22,24 @@ export default class PersonalCalendar extends React.Component<IPersonalCalendarP
...
+  /**
+   * Get timezone for logged in user
+   */
+  private _getTimeZone() {
+    return new Promise<string>((resolve, reject) => {
```
Comment (waldekmastykarz) [inline solution/src/webparts/personalCalendar/components/PersonalCalendar.tsx]: Could you add types information to explicitly specify type of data going in and out of the method?

**Comment by VesaJuvonen** [comment]
@Laskewitz - any change we could get the small adjustments on this as we would love to get your improvements on the actual solution?

**Comment by VesaJuvonen** [comment]
@Laskewitz - any change we could get the small adjustments on this as we would love to get your improvements on the actual solution? Thx.

**Comment by VesaJuvonen** [comment]
Thank you very much @Laskewitz on this change. I did eventually ended up doing the change manually based on the original code you provided with updated types. This is not included in the May release of the Starter Kit going out later today / tomorrow and you'll absolutely get acknowledged on the May notes as a contributor. 

Requested changes by @waldekmastykarz were small and as your contribution was valuable for others, processed this now manually. Closing this PR but code change is included in the code. Thanks for your contributions towards the community, highly appreciated.

**Pull request closed without merging**